Crash with metadata.nameX

v0.1.7 crashes with

```
apiVersion: v1
kind: Foo
metadata:
  nameX: foo
```
but works with
```
apiVersion: v1
kind: Foo
metadata:
  name: foo
```

```
panic: not a string

goroutine 1 [running]:
runtime/debug.Stack(0xc0001659e0, 0x5b86a0, 0x62a200)
	runtime/debug/stack.go:24 +0x9f
main.capturePanic()
	github.com/jrhouston/tfk8s/tfk8s.go:219 +0x76
panic(0x5b86a0, 0x62a200)
	runtime/panic.go:965 +0x1b9
github.com/zclconf/go-cty/cty.Value.AsString(0x0, 0x0, 0x0, 0x0, 0x754ec0, 0x3)
	github.com/zclconf/go-cty@v1.8.0/cty/value_ops.go:1251 +0x1a5
main.yamlToHCL(0x62f928, 0xc00016c120, 0x5c0ce0, 0xc00007d9b0, 0x0, 0x0, 0x0, 0x0, 0x62f928, 0xc00016c120, ...)
	github.com/jrhouston/tfk8s/tfk8s.go:110 +0x331
main.YAMLToTerraformResources(0x62d0a8, 0xc000010010, 0x0, 0x0, 0x0, 0x0, 0x20, 0xc000020b8a, 0xc000013c30)
	github.com/jrhouston/tfk8s/tfk8s.go:194 +0x41a
main.main()
	github.com/jrhouston/tfk8s/tfk8s.go:251 +0x35c


⚠️  Oh no! Looks like your manifest caused tfk8s to crash.
```

See https://github.com/argoproj/argo-workflows/blob/master/examples/archive-location.yaml for a real example.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,7 @@
+# 0.1.8
+
+- Support resources using generateName 
+
 # 0.1.7
 
 - Escape shell vars in HCL output

--- a/tfk8s.go
+++ b/tfk8s.go
@@ -107,10 +107,19 @@ func yamlToHCL(doc cty.Value, providerAlias string, stripServerSide bool, mapOnl
 		mm := doc.AsValueMap()
 		kind := mm["kind"].AsString()
 		metadata := mm["metadata"].AsValueMap()
-		name := metadata["name"].AsString()
 		var namespace string
 		if v, ok := metadata["namespace"]; ok {
 			namespace = v.AsString()
+		}
+
+		var name string
+		if n, ok := metadata["name"]; ok {
+			name = n.AsString()
+		} else if n, ok := metadata["generateName"]; ok {
+			name = n.AsString()
+			if name[len(name)-1] == '-' {
+				name = name[:len(name)-1]
+			}
 		}
 
 		resourceName := kind

--- a/tfk8s_test.go
+++ b/tfk8s_test.go
@@ -40,6 +40,39 @@ resource "kubernetes_manifest" "configmap_test" {
 	assert.Equal(t, strings.TrimSpace(expected), strings.TrimSpace(output))
 }
 
+func TestYAMLToTerraformResourcesGenerateName(t *testing.T) {
+	yaml := `---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  generateName: test-name-
+data:
+  TEST: test`
+
+	r := strings.NewReader(yaml)
+	output, err := YAMLToTerraformResources(r, "", false, false)
+
+	if err != nil {
+		t.Fatal("Converting to HCL failed:", err)
+	}
+
+	expected := `
+resource "kubernetes_manifest" "configmap_test_name" {
+  manifest = {
+    "apiVersion" = "v1"
+    "data" = {
+      "TEST" = "test"
+    }
+    "kind" = "ConfigMap"
+    "metadata" = {
+      "generateName" = "test-name-"
+    }
+  }
+}`
+
+	assert.Equal(t, strings.TrimSpace(expected), strings.TrimSpace(output))
+}
+
 func TestYAMLToTerraformResourcesEscapeShell(t *testing.T) {
 	yaml := `---
 apiVersion: v1

